### PR TITLE
Added AWS IAM data helpers

### PR DESCRIPTION
**Pull request by nfx**
End-to-end example of provisioning Cross-account IAM role:

```hcl
variable "account_id" {
  type        = string
  description = "External ID you find on https://accounts.cloud.databricks.com/#aws"
}

data "databricks_aws_crossaccount_policy" "this" {}
resource "aws_iam_policy" "cross_account_policy" {
  name   = "${var.prefix}-crossaccount-iam-policy"
  policy = data.databricks_aws_crossaccount_policy.this.json
}

data "databricks_aws_assume_role_policy" "this" {
    external_id = var.account_id
}

resource "aws_iam_role" "cross_account" {
  name               = "${var.prefix}-crossaccount-iam-role"
  assume_role_policy = data.databricks_aws_assume_role_policy.this.json
  description        = "Grants Databricks full access to VPC resources"
}

resource "aws_iam_role_policy_attachment" "cross_account" {
  policy_arn = aws_iam_policy.cross_account_policy.arn
  role       = aws_iam_role.cross_account.name
}

// required only in case of multiworkspace setup
resource "databricks_mws_credentials" "this" {
  provider         = databricks.mws
  account_id       = var.account_id
  credentials_name = "${var.prefix}-creds"
  role_arn         = aws_iam_role.cross_account.arn
}
```

### CODE REVIEW

File: access/data_aws_policies.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+	Effect       string                       `json:"Effect,omitempty"`
+	Actions      interface{}                  `json:"Action,omitempty"`
+	NotActions   interface{}                  `json:"NotAction,omitempty"`
+	Resources    interface{}                  `json:"Resource,omitempty"`
+	NotResources interface{}                  `json:"NotResource,omitempty"`
+	Principal    map[string]string            `json:"Principal,omitempty"`
+	Condition    map[string]map[string]string `json:"Condition,omitempty"`
+}
+
+func DataAwsCrossAccountRolicy() *schema.Resource {
```
Comment (stikkireddy) [inline access/data_aws_policies.go:27]: This needs a docs page

File: docs/data-sources/aws_assume_role_policy.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// required only in case of multiworkspace setup
+resource "databricks_mws_credentials" "this" {
+  provider         = databricks.mws
+  account_id       = var.account_id
+  credentials_name = "${var.prefix}-creds"
+  role_arn         = aws_iam_role.cross_account.arn
+}
+```
+
+## Attribute Reference
```
Comment (stikkireddy) [inline docs/data-sources/aws_assume_role_policy.md:46]: All the other docs make a clear distinction between attribute and argument. 

Argument should be user inputs where attributes are outputs/read only values computed by the resource/datasource. 

This should use the naming convention 
## Argument reference
* `<varname>` - (Required/Computed/Optional) (Datatype) (content)

## Attribute reference
* `<varname>` - (Datatype) (content)

File: docs/data-sources/aws_bucket_policy.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+data "databricks_aws_bucket_policy" "stuff" {
+  bucket_name = aws_s3_bucket.this.bucket
+}
+
+resource "aws_s3_bucket_policy" "this" {
+  bucket     = aws_s3_bucket.this.id
+  policy     = data.databricks_aws_bucket_policy.this.json
+}
+```
+## Argument Reference
```
Comment (stikkireddy) [inline docs/data-sources/aws_bucket_policy.md:24]: All the other docs make a clear distinction between attribute and argument. 

Argument should be user inputs where attributes are outputs/read only values computed by the resource/datasource. 

The bucket_name should be an argument where the json is an attribute.

This should use the naming convention 
## Argument reference
* `<varname>` - (Required/Computed/Optional) (Datatype) (content)

## Attribute reference
* `<varname>` - (Datatype) (content)

**Comment by nfx** [comment]
@stikkireddy will add documentation as requested in pre-release testing branch

**Pull request merged**